src/main/java/com/main/fastserver/Scheduled/ProcessService.java

```diff
@@ -0,0 +1,131 @@
+package com.main.fastserver.Scheduled;
+
+import com.main.fastserver.Entity.Domain;
+import com.main.fastserver.Entity.Skill;
+import com.main.fastserver.Entity.SubDomain;
+import com.main.fastserver.Service.DomainService;
+import com.main.fastserver.Service.SkillService;
+import com.main.fastserver.Service.SubDomainService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Configurable;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class ProcessService {
+
+    private static final Logger log = LoggerFactory.getLogger(ScheduledTask.class);
+
+    @Autowired
+    private DomainService domainService;
+
+    @Autowired
+    private SubDomainService subDomainService;
+
+    @Autowired
+    private SkillService skillService;
+
+    public void process(Map<String, Map<String, Map<String, Skill>>> domainsMap) {
+        List<Domain> domainsInDataBase = domainService.findAll();
+        this.processDelete(domainsMap, domainsInDataBase);
+        for(String titleDomain : domainsMap.keySet()) {
+            boolean isPresent = false;
+            for(Domain domainInDatabase : domainsInDataBase) {
+                if(titleDomain.equals(domainInDatabase.getTitle())) {
+                    isPresent = true;
+                    List<SubDomain> subDomains = domainInDatabase.getSubdomains();
+                    for(String titleSubDomain : domainsMap.get(titleDomain).keySet()) {
+                        boolean isPresentSubDomain = false;
+                        if(subDomains != null) {
+                            for (SubDomain subDomain : subDomains) {
+                                if (titleSubDomain.equals(subDomain.getTitle())) {
+                                    isPresentSubDomain = true;
+                                    List<Skill> skills = subDomain.getSkills();
+                                    for(String titleSkill : domainsMap.get(titleDomain).get(titleSubDomain).keySet()) {
+                                        boolean isPresentSkill = false;
+                                        if(skills != null) {
+                                            for(Skill skill : skills) {

```

Comment:
6 boucles for ????? What ? Pourquoi ?
On avait pas dit on parcours une fois d'un côté avec 3 boucles for.
Et on reparcourait ensuite une fois dans l'autre sens avec 3 boucles for ?

Jamais on doit en avoir 6.